src/routes/safe/components/Apps/ManageApps.js

```diff
@@ -0,0 +1,183 @@
+// @flow
+import { ButtonLink, Checkbox, ManageListModal, Text, TextField } from '@gnosis.pm/safe-react-components'
+import React, { useState } from 'react'
+import { FormSpy } from 'react-final-form'
+import styled from 'styled-components'
+
+import { getAppInfoFromUrl } from './utils'
+
+import Field from '~/components/forms/Field'
+import DebounceValidationField from '~/components/forms/Field/DebounceValidationField'
+import GnoForm from '~/components/forms/GnoForm'
+import { composeValidators, required } from '~/components/forms/validator'
+import Img from '~/components/layout/Img'
+import appsIconSvg from '~/routes/safe/components/Transactions/TxsTable/TxType/assets/appsIcon.svg'
+
+const FORM_ID = 'add-apps-form'
+
+const StyledText = styled(Text)`
+  margin-bottom: 19px;
+`
+
+const StyledTextFileAppName = styled(TextField)`
+  && {
+    width: 335px;
+  }
+`
+
+const AppInfo = styled.div`
+  margin: 36px 0 24px 0;
+
+  img {
+    margin-right: 10px;
+  }
+`
+
+const StyledCheckbox = styled(Checkbox)`
+  margin: 0;
+`
+const APP_INFO = { iconUrl: appsIconSvg, name: '', error: false }
+
+type Props = {
+  appList: Array<{
+    id: string,
+    iconUrl: string,
+    name: string,
+    disabled: boolean,
+  }>,
+  onAppAdded: (app: any) => void,
+  onAppToggle: (appId: string, enabled: boolean) => void,
+}
+
+const ManageApps = ({ appList, onAppAdded, onAppToggle }: Props) => {
+  const [isOpen, setIsOpen] = useState(false)
+
+  const [appInfo, setAppInfo] = useState(APP_INFO)
+  const [isSubmitDisabled, setIsSubmitDisabled] = useState(true)
+
+  const onItemToggle = (itemId: string, checked: boolean) => {
+    onAppToggle(itemId, checked)
+  }
+
+  const handleSubmit = () => {
+    setIsOpen(false)
+    onAppAdded(appInfo)
+  }
+
+  const cleanAppInfo = () => setAppInfo(APP_INFO)
+
+  const urlValidator = (value: string) => {
```

Comment:
this function doesn't have to be defined inside the component

Reply:
done